### PR TITLE
Corrections in the release checklist.

### PR DESCRIPTION
**Pull request by calavera**
- Add a note about creating the release in GitHub.
- Fix docs release process.

Signed-off-by: David Calavera david.calavera@gmail.com

### CODE REVIEW

File: project/RELEASE-CHECKLIST.md [hunk 1]
```diff
@@ -351,6 +351,15 @@ git tag -a $VERSION -m $VERSION bump_$VERSION
 git push origin $VERSION
 ```
 
+Once the tag is pushed, go to GitHub and create a [new release](https://github.com/docker/docker/releases/new).
+If the tag is for an RC make sure you check `This is a pre-release` at the bottom of the form.
+
+Select the tag that you just pushed as version and paste the changelog in the description of the release.
```
Comment (duglin) [inline project/RELEASE-CHECKLIST.md]: a/as version/as the version/

File: project/RELEASE-CHECKLIST.md [hunk 2]
```diff
@@ -358,25 +367,12 @@ branch afterwards!
 
 ### 14. Update the docs branch
 
-If this is a MAJOR.MINOR.0 release, you need to make an branch for the previous release's
-documentation:
+You will need to point the docs branch to the released tag:
```
Comment (duglin) [inline project/RELEASE-CHECKLIST.md]: perhaps:  `... to the newly created release tag:`   if I'm following things.

**Comment by calavera** [comment]
@duglin fixed, thanks.

**Comment by duglin** [comment]
LGTM

**Comment by SvenDowideit** [comment]
LGTM

**Pull request merged**